ledger/complete/wal/checkpointer.go

```diff
@@ -1086,3 +1086,35 @@ func CopyCheckpointFile(filename string, from string, to string) (
 
 	return newPaths, nil
 }
+
+// SoftlinkCheckpointFile creates soft links of the checkpoint file including the part files from the given `from` to
+// the `to` directory
+func SoftlinkCheckpointFile(filename string, from string, to string) ([]string, error) {
```

Comment:
Why not just link the whole directory instead of individual files?

Reply:
If we link the entire directory, all checkpoint will be stored to the bootstrap/execution-state folder. 

It would break if we would like to re-execute from root block, which we usually just delete the data folder, including the old checkpoint files in it. However, if we link the whole directory, then deleting the data folder won't delete those old checkpoint files. They will be used to bootstrap the node, which will fail.